public/push-messaging-sw.js

```diff
@@ -0,0 +1,30 @@
+/* global importScripts firebase FCM_SENDER_ID */
+// Give the service worker access to Firebase Messaging.
+// Note that you can only use Firebase Messaging here, other Firebase libraries
+// are not available in the service worker.
+
+importScripts('/lib/firebase/firebase-app.js');
+importScripts('/lib/firebase/firebase-messaging.js');
+importScripts('/config/sw.js');
+
+firebase.initializeApp({
+  'messagingSenderId': FCM_SENDER_ID
+});
+
+var messaging = firebase.messaging();
+
+messaging.setBackgroundMessageHandler(function(payload) {
+  // not actually used, but without it here firefox does not receive messages...
+  console.log('received payload', payload);
```

Comment:
This function _never_ gets fired but it has to be there? `messaging.setBackgroundMessageHandler(Function.prototype)` might be handy.

Also if Angular is already in `window` at this point, [angular.noop](https://docs.angularjs.org/api/ng/function/angular.noop) is available.